concourse/pipelines/modelbase.py

```diff
@@ -69,10 +70,27 @@ def custom_init(self, raw_dict: dict):
         if raw_dict.get('output_dir', None):
             name = raw_dict['output_dir']
             self.add_output(name + '_path', name + '_path')
+        # normalise step-arguments, if given.
+        arguments = raw_dict.get('arguments', None)
+        if arguments:
+            if isinstance(arguments, dict):
+                raise ModelValidationError('step arguments must be scalar values or lists containing only scalar values')
+            if not isinstance(arguments, list):
+                arguments = [arguments]
+            if any(isinstance(t, ()))
+            raw_dict['arguments'] = [str(argument) for argument in arguments]
 
     def image(self):
         return self.raw.get('image', None)
 
+    def arguments(self):
+        arguments = self.raw.get('arguments', None)
+        if arguments is None:
+            return arguments
+        if len(arguments) == 1:
```

Comment:
no! this way, you would erroneously handle a single-elemented list the same way as a scalar